sound/soc/sof/core.c

```diff
@@ -291,6 +291,8 @@ static int sof_probe(struct platform_device *pdev)
 		goto ipc_err;
 	}
 
+/* bypass DSP if in force legacy hda debug mode */
+#ifndef CONFIG_SND_SOC_SOF_FORCE_LEGACY_HDA
```

Comment:
or DSP is not enabled, you could test this on SKL!

Reply:
OK, I will add checking of DSP's availability.